### PR TITLE
[RDF] Prevent compiler warnings with clang and range-based loop over …

### PR DESCRIPTION
**Pull request by stwunsch**
…vector<bool>

### CODE REVIEW

File: tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx [hunk 1]
```diff
@@ -462,9 +462,12 @@ public:
...
       for (unsigned int i = 1; i < fColls.size(); ++i) {
-         auto &coll = fColls[i];
-         for (auto &&v : *coll) {
-            FillColl(v, *rColl);
+         const auto &coll = fColls[i];
+         const auto begin = coll->begin();
+         // Use an explicit loop here to prevent compiler warnings introduced by
+         // clang's range-based loop analysis and vector<bool> references.
+         for (std::size_t j = 0; j < coll->size(); j++) {
+            FillColl(*std::next(begin, j), *rColl);
```
Comment (bluehood) [inline tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx]: I think it's more performant to increment the iterator in the for loop and avoid calls to `next` and `size`:

```c++
const auto end = coll->end(); 
for (auto it = coll->begin(); it != end; ++it)
   FillColl(*it, ...);
```

Comment (bluehood) [reply tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx]: also `cbegin` and `cend` would be (admittedly very slightly) better

Comment (stwunsch) [reply tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx]: Absolutely! Thanks :)

**Comment by stwunsch** [comment]
I've tested on the mac1014/clang100 build node and this fixes the warnings.

**Comment by stwunsch** [comment]
@phsft-bot build

**Pull request merged**